src/content/app/tools/vep/views/vep-submission-results/VepSubmissionResults.tsx

```diff
@@ -0,0 +1,350 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useVepResultsQuery } from 'src/content/app/tools/vep/state/vep-api/vepApiSlice';
+
+import { Table, ColumnHead } from 'src/shared/components/table';
+import VariantConsequence from 'src/shared/components/variant-consequence/VariantConsequence';
+import VepResultsGene from './components/vep-results-gene/VepResultsGene';
+import VepResultsLocation from './components/vep-results-location/VepResultsLocation';
+
+import type {
+  VEPResultsResponse,
+  AlternativeVariantAllele,
+  PredictedTranscriptConsequence
+} from 'src/content/app/tools/vep/types/vepResultsResponse';
+
+import styles from './VepSubmissionResults.module.css';
+
+/**
+ * TODO:
+ * - Add unique id to variants after they are requested (to use for keys)
+ * - Enable expansion/collapsing of transcripts of a given gene
+ * - Show intergenic variants as predicted consequences alongside transcript consequences
+ * - Consider pagination (should it be part of url?)
+ * - Make the table scrollable
+ */
+
+const VepSubmissionResults = () => {
+  const { currentData: vepResults } = useVepResultsQuery();
+
+  return (
+    <div className={styles.container}>
+      <div>Vep analysis</div>
+      <div className={styles.resultsBox}>
+        <div>Area above the table</div>
+        {vepResults && <VepResultsTable variants={vepResults.variants} />}
+      </div>
+    </div>
+  );
+};
+
+const VepResultsTable = (props: {
+  variants: VEPResultsResponse['variants'];
+}) => {
+  const { variants } = props;
+
+  return (
+    <Table className={styles.table}>
+      <thead>
+        <tr>
+          <ColumnHead>Variant</ColumnHead>
+          <ColumnHead>Ref</ColumnHead>
+          <ColumnHead>Location</ColumnHead>
+          <ColumnHead>Alt allele</ColumnHead>
+          <ColumnHead>Genes</ColumnHead>
+          <ColumnHead>Transcripts</ColumnHead>
+          <ColumnHead>Consequences</ColumnHead>
+        </tr>
+      </thead>
+      <tbody>
+        {/* Use something more reliable for key than index */}
```

Comment:
Can a rsID be used for this?

Reply:
No; the variant name is defined (or omitted) by the user. It is not guaranteed to be unique (in a VCF input, if user does not define it, it is represented with a dot).